### PR TITLE
Heretic blades can properly butcher

### LINKED ISSUE #68846: the sound when you start to butcher a mob is gone
## Reproduction:

1. Have an animal carcass that butchers when you hit it with a knife.
2. Touch with knife and watch in astonishment when it is completely silent.

### LINKED ISSUE #73122: Heretic blades are unable to produce hides/meat when butchering
Reporting client version: 514.1589

## Round ID:
[199299](https://scrubby.melonmesa.com/round/199299)

## Reproduction:
After butchering a corgi I was given the message
"You fail to harvest some of the corgi meat from the corgi.
x3

You fail to harvest some of the corgi hide from the corgi."

Testing on local showed this was the same for other animals.
This is impactful because the codex cicatrix craft requires a hide and most of the time the only sharp item a heretic will be using will be their knife.

### PR DESCRIPTION
**Pull request by the-orange-cow**
## About The Pull Request

Presets initial butchering component, which allows heretic blades to properly butcher.

## Why It's Good For The Game

Fixes https://github.com/tgstation/tgstation/issues/73122
Fixes https://github.com/tgstation/tgstation/issues/68846

## Changelog

:cl:
fix: heretic blades can now properly butcher
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
> Actually it seems like the issue is a bit deeper than just heretic blades
> 
> Investigating further:
> 
> * All sharp items with decent force are able to butcher
> 
> https://github.com/tgstation/tgstation/blob/f6f3e9498fd554c43778e959e2612709147a8905/code/game/objects/items.dm#L225-L226
> 
> * The default set butchering effectiveness is 100
> 
> https://github.com/tgstation/tgstation/blob/f6f3e9498fd554c43778e959e2612709147a8905/code/datums/components/butchering.dm#L4-L5
> 
> * However, butchering `initialize` does not use any default arguments, meaning all the default values in the component declaration are reset to `null`
> 
> https://github.com/tgstation/tgstation/blob/f6f3e9498fd554c43778e959e2612709147a8905/code/datums/components/butchering.dm#L17-L29
> 
> I believe the initialize arguments need to start pre-filled with the default arguments; right now it's setting `effectiveness` to `null`, which is expected to be a number.
> 
> However this in turn raises the question "should all sharp objects be able to butcher effectively?"
> 
> (In my opinion, sharp objects should have a default effectiveness of like 50. Then on a per-item basis, like for heretic blades, it can be upped to 100 or beyond.)

Ah, thank you for explaining the issue.  This seems very far above my head, so I'll close this pull, as I likely won't be capable of implementing an appropriate fix.

**Comment by the-orange-cow** [comment]
> I think you're undershooting yourself, it'd be as simple as...
> 
> ```
> /datum/component/butchering/Initialize(
> 	speed = 8 SECONDS,
> 	effectiveness = 100,
> 	bonus_modifier = 0,
> 	butcher_sound = 'sound/effects/butcher.ogg',
> 	disabled = FALSE,
> 	can_be_blunt = FALSE,
> 	butcher_callback,
> )
> ```

Thank you for explaining how to implement the needed adjustments!

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 1]
```diff
@@ -21,6 +21,13 @@
 	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "rend")
 	var/after_use_message = ""
 
+/obj/item/melee/sickly_blade/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/butchering, \
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:26]: With the new code you added, is this needed still?
It should be automatically applied now, and the vars are just the default so it seems like this just isn't needed

Comment (the-orange-cow) [reply code/modules/antagonists/heretic/items/heretic_blades.dm]: I tested it out, and you are correct, it still properly butchers without the new code.

**Pull request merged**